servers/quarkus-server/src/main/java/org/projectnessie/server/catalog/CatalogProducers.java

```diff
@@ -109,37 +106,28 @@ public void closeSdkHttpClient(@Disposes @CatalogS3Client SdkHttpClient client)
     client.close();
   }
 
-  @SuppressWarnings({"unchecked", "rawtypes", "UnnecessaryLocalVariable"})
   @Produces
   @Singleton
-  public S3Options<S3BucketOptions> s3Options(CatalogS3Config s3Config) {
-    S3Options opts = s3Config;
-    S3Options<S3BucketOptions> r = opts;
-    return r;
+  public S3ProgrammaticOptions s3Options(CatalogS3Config s3Config) {
+    return S3ProgrammaticOptions.copyOf(s3Config);
   }
 
-  @SuppressWarnings({"unchecked", "rawtypes", "UnnecessaryLocalVariable"})
   @Produces
   @Singleton
-  public GcsOptions<GcsBucketOptions> gcsOptions(CatalogGcsConfig gcsConfig) {
-    GcsOptions opts = gcsConfig;
-    GcsOptions<GcsBucketOptions> r = opts;
-    return r;
+  public GcsProgrammaticOptions gcsOptions(CatalogGcsConfig gcsConfig) {
+    return GcsProgrammaticOptions.copyOf(gcsConfig);
```

Comment:
```suggestion
    return GcsProgrammaticOptions.copyOf(gcsConfig).normalizeBuckets();
```

Can we have Quarkus tests to validate this?

Reply:
I modified `ObjectStorageMockTestResourceLifecycleManager` to exercise buckets with custom names.